consensus/tendermint/rule_skip_round_test.go

```diff
@@ -0,0 +1,51 @@
+package tendermint
+
+import "testing"
+
+func TestSkipRound(t *testing.T) {
```

Comment:
Just a suggestion to further improve readability
```suggestion
func TestSkipRound(t *testing.T) {
	thisNode := 3
	peer1:=0
	peer2:=1
	
```

Reply:
```
func TestSkipRound(t *testing.T) {
	thisNode := 3
	peer1:=0
	peer2:=1

	expectedHeight := height(0)
	rPrime, rPrimeVal := round(4), value(10)
	round4Value := value(10)

	t.Run("Line 55 (Proposal): Start round r' when f+1 future round messages are received from round r'", func(t *testing.T) {
		stateMachine := setupStateMachine(t, 4, thisNode)
	
		futureRound := newTestRound(t, stateMachine, expectedHeight, rPrime)

		stateMachine.processStart(round(0))
		futureRound.validator(peer1).prevote(&rPrimeVal)
		futureRound.validator(peer2).proposal(rPrimeVal, -1)

		// The step is not propose because the proposal which is received in round r' leads to consensus
		// engine broadcasting prevote to the proposal which changes the step from propose to prevote.
		assertState(t, stateMachine, expectedHeight, rPrime, prevote)
	})
```

Reply:
I think `thisNode` makes sense.
Use `peer1` for validator 0 may cause confusion, while `peer0` is redundant.

Reply:
By confusion, I meant switching between 0-based and 1-based index

Reply:
How about
```

  futureRound.peer(0).prevote(&rPrimeVal)
        futureRound.peer(1).proposal(rPrimeVal, -1

```
Another approach is to replace 0-based index by 1-based index

Reply:
> I think thisNode makes sense.
> Use peer1 for validator 0 may cause confusion, while peer0 is redundant.

What about just using `peer0` and `peer1`?

Reply:
It just seems like an easier update than renaming validator to peer. Plus I think validator is a better name. I don't feel strongly either way, so I'll let you make the decision :)

Reply:
I'd keep `validator(0)` for now, because I think it's more concise than `validator(peer0)`.